src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -40,10 +40,23 @@ public class FhirCosmosSearchService : SearchService
             SearchOptions searchOptions,
             CancellationToken cancellationToken)
         {
-            return await ExecuteSearchAsync(
+            SearchResult searchResult = await ExecuteSearchAsync(
                 _queryBuilder.BuildSqlQuerySpec(searchOptions),
                 searchOptions,
                 cancellationToken);
+
+            if (searchOptions.IncludeTotal == TotalType.Accurate && !searchOptions.CountOnly)
+            {
+                searchOptions.CountOnly = true;
```

Comment:
I think I'd prefer this manipulation to be done in a method on SearchOptions for now. i.e. `searchOptions.AsCountOnly()`. The property can then remain as private.

Reply:
Feels a bit strange adding a set method to a class of properties with internal setters, but I understand that manipulating CountOnly directly in these files isn't super nice. Let me know how 809259a looks.